demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -868,6 +989,8 @@ static int unsubscribeFromTopic( MQTTContext_t * pMqttContext )
         LogInfo( ( "UNSUBSCRIBE sent for topic %.*s to broker.\n\n",
                    MQTT_EXAMPLE_TOPIC_LENGTH,
                    MQTT_EXAMPLE_TOPIC ) );
+        /* Reset the global suback status variable. */
+        globalSubAckStatus = MQTTSubAckFailure;
```

Comment:
The reset of this variable in the unsubscribe call seems out-of-place. It can probably be avoided.

Reply:
It should instead be reset after every subscription request cycle (in `publishSubscribeLoop` and the `handleResubscribe`)